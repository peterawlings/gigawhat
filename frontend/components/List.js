import React from "react";

const List = ({ list }) => {
  return (
    <div>
      <ul>
        {list.map(listing => (
          <li>{listing.venue}</li>
        ))}
      </ul>
    </div>
  );
};

export default List;
