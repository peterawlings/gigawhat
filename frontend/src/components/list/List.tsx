import React, { FunctionComponent } from "react";

interface gigData {
  id: number;
  genre: Array<string>;
  artistName: string;
  address: {
    street: string;
    suburb: string;
    postCode: number;
  };
  price: number;
  venue: string;
}

interface gigListProps {
  list: Array<gigData>;
}

const List: FunctionComponent<gigListProps> = ({ list }) => {
  return (
    <div>
      <ul data-testid="gigList-ul">
        {list.map((listing: any) => (
          <li key={listing.id}>{`${listing.venue} + ${listing.artistName}`}</li>
        ))}
      </ul>
    </div>
  );
};

export default List;
