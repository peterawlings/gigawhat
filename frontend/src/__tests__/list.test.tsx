import React from "react";
import { render, cleanup } from "@testing-library/react";
import List from "../components/list";

const defaultProps = {
  list: [
    {
      id: 0,
      genre: ["acoustic"],
      artistName: "Duan and only",
      address: {
        street: "124 Seigel Street",
        suburb: "CBD",
        postCode: 2487,
      },
      price: 30,
      venue: "Ivanhoe",
    },
    {
      id: 1,
      genre: ["funk"],
      artistName: "Rolf",
      address: {
        street: "961 Plymouth Street",
        suburb: "Manly",
        postCode: 2185,
      },
      price: 10,
      venue: "Ivanhoe",
    },
  ],
};

afterEach(cleanup);

describe("<List/>", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<List {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should show all gig listings", () => {
    const { getByTestId } = render(<List {...defaultProps} />);
    const gigList = getByTestId("gigList-ul");
    expect(gigList.children.length).toBe(2);
  });
});
