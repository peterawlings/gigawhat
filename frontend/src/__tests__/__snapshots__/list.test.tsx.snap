// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> renders correctly 1`] = `
<DocumentFragment>
  <div>
    <ul
      data-testid="gigList-ul"
    >
      <li>
        Ivanhoe + Duan and only
      </li>
      <li>
        Ivanhoe + Rolf
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
